mantidproject/mantid
# CORELLI tube calibration: switch acceptance criterion

**Description of work.**
- [x] criterion function using the vertical (Y-coordinate) fitted positions of the wire shadows
- [x] switch criteria functions (from pixel to vertical positions) in all dependent functions


**To test:**
First make sure that the `scripts/` directory located under the Mantid source repository is added to the list of scripts directories. Then, in the script Editor of the workbench, run the following script (should take not more than 5 minutes):
```
from corelli.calibration.bank import calibrate_banks
from corelli.calibration.utils import apply_calibration, load_banks
counts_file = '/SNS/CORELLI/shared/tmp/CORELLI_124023_counts.nxs'
load_banks(counts_file, bank_selection='10-19', output_workspace='counts')
calibrate_banks('counts', bank_selection='10-19')
apply_calibration('counts', 'calib10', output_workspace='counts_10')
apply_calibration('counts', 'calib14', output_workspace='counts_14')
```
Open the instrument view for workspaces `counts` and `counts_10`, then look at `bank10` and verify that all tubes in this bank have been calibrated for `counts_10`. Before switching criterion, `tube13` [was not successfully calibrated](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/121).

<img width="400" alt="Capture" src="https://user-images.githubusercontent.com/1136006/96651727-6aa59c00-1303-11eb-9c58-4f5d72cf7429.PNG">

Open the instrument view for workspaces `counts` and `counts_10`, then look at `bank10` and verify that all tubes in this bank have been calibrated for `counts_10`


Fixes #29766





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
